### PR TITLE
feat: add pending to role filter

### LINKED ISSUE #13810: [CAL-3188] [CAL-3187] improvements to org member list
- [ ] sticky actionbar in wrong place
CleanShot 2024-02-21 at 19 27 50@2x

- [ ] make teams clickable and filterable, same as roles
CleanShot 2024-02-21 at 19 25 49@2x

- [ ] thereby, change "Roles" to "Filters" because we are adding Teams as another filter

CleanShot 2024-02-21 at 19 26 19@2x

- [ ] Change placeholder to "Filter"
- [ ] Add divider after roles and add teams
CleanShot 2024-02-21 at 19 26 55@2x

- [ ] add total members (only if it doenst slow us down like crazy)
CleanShot 2024-02-21 at 19 24 51@2x

- [ ] add pending here:
CleanShot 2024-02-21 at 19 01 52@2x

- [ ] checkboxes not aligned
image

<sub>[CAL-3188](https://linear.app/calcom/issue/CAL-3188/[cal-3187]-improvements-to-org-member-list)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

Fixes #13810

Continuation  #13859

https://github.com/calcom/cal.com/assets/74371312/1745dfa3-2e78-411c-8561-900d47e397f4

## Type of change

- New feature (non-breaking change which adds functionality)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/users/components/UserTable/UserListTable.tsx [hunk 1]
```diff
@@ -207,6 +207,17 @@ export function UserListTable() {
...
         },
         filterFn: (rows, id, filterValue) => {
+          if (filterValue.length === 1 && filterValue[0] === "PENDING") {
+            // Show only pending members
+            return !rows.original.accepted;
+          }
+
+          if (filterValue.length > 1 && filterValue.includes("PENDING")) {
+            // Show pending members and the selected roles
+            return !rows.original.accepted && filterValue.includes(rows.getValue(id));
```
Comment (Udit-takkar) [inline packages/features/users/components/UserTable/UserListTable.tsx:217]: I think you meant to use OR here instead of &&

Comment (Amit91848) [reply packages/features/users/components/UserTable/UserListTable.tsx]: I thought showing role AND pending would be better

i.e if you check admin and pending boxes, you show admins that are pending

Comment (Udit-takkar) [reply packages/features/users/components/UserTable/UserListTable.tsx]: but all other checkboxes doesn't work like this. I also tried selection owner and members and it returned me both members and owner.

Comment (Udit-takkar) [reply packages/features/users/components/UserTable/UserListTable.tsx]: I think OR is fine for now. I'll change it later if we don't need it.

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**